services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/S3CrossRegionAsyncClient.java

```diff
@@ -52,23 +52,17 @@ protected <T extends S3Request, ReturnT> CompletableFuture<ReturnT> invokeOperat
 
         AwsRequestOverrideConfiguration overrideConfiguration = updateUserAgentInConfig(request);
         T userAgentUpdatedRequest = (T) request.toBuilder().overrideConfiguration(overrideConfiguration).build();
-
         if (!bucket.isPresent()) {
             return operation.apply(userAgentUpdatedRequest);
         }
         String bucketName = bucket.get();
 
         CompletableFuture<ReturnT> returnFuture = new CompletableFuture<>();
-        CompletableFuture<ReturnT> apiOperationFuture = bucketToRegionCache.containsKey(bucketName) ?
-                                                        operation.apply(
-                                                            requestWithDecoratedEndpointProvider(
-                                                                userAgentUpdatedRequest,
-                                                                () -> bucketToRegionCache.get(bucketName),
-                                                                serviceClientConfiguration().endpointProvider().get()
-                                                            )
-                                                        ) :
-                                                        operation.apply(userAgentUpdatedRequest);
-
+        CompletableFuture<ReturnT> apiOperationFuture = operation.apply(
```

Comment:
Why are we removing `bucketToRegionCache.containsKey(bucketName)` check?

Reply:
Earlier, for the first calls, since CachedBucket was null, we used to skip overriding the EndpointProvider because we were making the calls as they were. Now, since we need to update the 'useGlobalEndpoint' even for the first call, we need to override the EndpointProvider